src/scripts/convert_lovd2.php

```diff
@@ -0,0 +1,1312 @@
+    $aPhenotypes = array();
+
+    // Array for storing unique variant_id/patient_id combinations as key. This
+    // is used to find homozygous variants as they will occur twice in the
+    // LOVD2 export file with identical variant_id/patient_id.
+    $aUniqueRecords = array();
+
+    $nCounter = 0;
+    foreach ($aData as $i => $sLine) {
+        // Set progress bar (leave 1 percent for output generation).
+        $oProgressBar->setProgress((++$nCounter / $nNumLines) * 99);
+        $oProgressBar->setMessage('Converting record ' . strval($nCounter) . ' of ' .
+            strval($nNumLines) . '...');
+
+        $sLine = trim($sLine);
+        if (empty($sLine) || $sLine[0] == "#" || preg_match('/^"?{{.*/', $sLine)) {
+            // Ignore blank lines, comments and the header line.
+            continue;
+        }
+
+        // Loop over fields in record and convert values according to $aFieldLinks.
+        $aInputRecord = explode("\t", $sLine);
+        $aRecord = array();
+        for ($i = 0; $i < count($aInputRecord); $i++) {
+            $sFieldName = $aInputHeaders[$i];
+            $sFieldValue = $aInputRecord[$i];
+            if (isset($aFieldLinks[$sFieldName])) {
+                if (count($aFieldLinks[$sFieldName]) == 2) {
+                    $aRecord[] = $sFieldValue;
+                } else {
+                    $aRecord[] = call_user_func($aFieldLinks[$sFieldName][2], $sFieldValue);
+                }
+            } else {
+                // Copy field as is.
+                $aRecord[] = $sFieldValue;
+            }
+        }
+
+        // Handle multiple observations for single variant in one patient
+        // (homozygous).
+        $sRecordID = $aInputRecord[array_search('ID_variantid_', $aInputHeaders)] . '_' .
+                     $aInputRecord[array_search('ID_patientid_', $aInputHeaders)];
+        if (array_key_exists($sRecordID, $aUniqueRecords)) {
+            // Combination variant_id/patient_id already seen, set previous
+            // record allele field to 3 (homozygous). Skip this record.
+            $aVOGRecords[$sRecordID]['allele'] = 3;
+            continue;
+        } else {
+            // Store current variant/patient combination for future reference.
+            $aUniqueRecords[$sRecordID] = null;
+        }
+
+        // Get submitter ID.
+        $sSubmitterID = null;
+        if (($i = array_search('ID_submitterid_', $aInputHeaders)) !== false) {
+            $sSubmitterID = $aRecord[$i];
+        }
+
+
+        // Create new disease if necessary.
+        $aDisease = lovd_getRecordForHeaders($aOutputHeaders['disease'], $aRecord,
+            $aSections['disease']);
+        if (!lovd_empty($aDisease['name'])) {
+            list($aDisease['id'], $bCreateNewDisease) = lovd_getDiseaseID($aDisease['name']);
+            if ($aDisease['id'] === false) {
+                // Something went wrong determining the disease, stop parsing.
+                break;
```

Comment:
Why stop parsing here? It would be more useful to continue, so the user will see all the errors in the entire file, instead of having to solve the problems one by one and retrying the conversion each time.